### ISSUE TITLE
TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by duckscofield** [reactjs/react-docgen#285]
I could understand this meaning. I follow JSDOC spec. But the page could not render 'PROPS & METHODS' block for display.

console.log
`It usually means that react-docgen does not understand your source code, try to file an issue here:
https://github.com/reactjs/react-docgen/issues`

**Comment by duckscofield** [comment]
OK, I give some codes as belows:

```
  /**
   * handle input value when typing
   * @param  {[type]} event [description]
   * @return {[type]}       [description]
   */
  handleInput = (event) => {
    const { value } = event.target;
    this.foundation_.handleInput_(value);
  }
```

if I use the arrow function or use static/get/set function, add comments, it will fail to parse

**Comment by danez** [comment]
Thanks for the example, I fixed the issue in #286. `{[type]}` was not supported.
